src/modules/PackageSlideShow.rb

```diff
@@ -188,150 +214,203 @@ def UpdateTotalProgress(_silent_check)
       UpdateTotalProgressValue()
     end
 
-    # Progress display update
-    # This is called via the packager's progress callbacks.
+    # For API backwards compatibility. DEPRECATED.
     #
-    # @param [Fixnum] pkg_percent  package percentage
+    def DisplayGlobalProgress
+      log.warning "DEPRECATED. Use UpdateTotalProgressText() instead."
+      UpdateTotalProgressText()
+    end
+
+    # Update the total progress text (not the value!).
+    #
+    def UpdateTotalProgressText
+      if @active_downloads > 0 && !parallel_download?
+        UpdateDownloadProgressText()
+      else
+        UpdateInstallationProgressText()
+      end
+    end
+
+    # Update the total progress text for downloading.
+    # This should only be used if parallel download + installation is not in effect.
     #
-    def UpdateCurrentPackageProgress(pkg_percent); end
+    def UpdateDownloadProgressText
+      SlideShow.SetGlobalProgressLabel(
+        _("Downloading...") +
+        Builtins.sformat(
+          # TRANSLATORS: This is about a remaining download size.
+          # %1 is the remaining size with a unit (kiB, MiB, GiB etc.),
+          # %2 the total download size, also with a unit.
+          _(" (Remaining: %1 of %2)"),
+          String.FormatSize(@expected_total_download_size - CurrentDownloadSize()),
+          String.FormatSize(@expected_total_download_size)
+        )
+      )
 
-    # update the download rate
-    def UpdateCurrentPackageRateProgress(_pkg_percent, _bps_avg, _bps_current)
       nil
     end
 
-    def DisplayGlobalProgress
-      rem_string = FormatRemainingSize(@total_size_to_install - @total_size_installed)
-
-      rem_string += ", " unless rem_string.empty?
+    # Update the total progress text for installing / updating / removing
+    # packages; or, for parallel download + installation, also for downloading.
+    #
+    def UpdateInstallationProgressText
+      installed_pkg = @installed_pkg_list.size
+      updated_pkg = @updated_pkg_list.size
+      remaining_string = FormatRemainingSize(@total_size_to_install - @total_installed_size)
+      remaining_string += ", " unless remaining_string.empty?
 
       SlideShow.SetGlobalProgressLabel(
-        Ops.add(
-          SlideShow.CurrentStageDescription,
-          Builtins.sformat(
-            _(" (Remaining: %1%2 packages)"),
-            rem_string,
-            @total_pkgs_to_install - @installed_packages - @updated_packages
-          )
+        SlideShow.CurrentStageDescription +
+        Builtins.sformat(
+          _(" (Remaining: %1%2 packages)"),
+          remaining_string,
+          @total_pkgs_to_install - installed_pkg - updated_pkg
```

Comment:
And I have the same feeling here. IMHO, it would be more understandable to have "Installating: 8/342 packages". On the other hand, a counter up is a mathematical representation of the pogress bar. So, maybe the remainder value is giving more information. I don't know.

Reply:
Let's discuss this live later today.

I don't have a strong opionion one way or the other; It was just like that before, and now we have two locations with very similar messages. We should be consistent.